docs/policy.md

```diff
@@ -0,0 +1,169 @@
+```
+app.js
+└─┬ some-package
+  └─┬ entropoetry
+    └── bn.js
+```
+
+`./lavamoat/node/policy.json`:
+```
+{
+  "resources": {
+    "some-package": {
+      "globals": {
+        "Buffer.from": true
+      },
+      "packages": {
+        "some-package>entropoetry": true
+      }
+    },
+    "some-package>entropoetry": {
+      "builtin": {
+        "assert": true,
+        "buffer.Buffer": true,
+        "zlib": true
+      },
+      "globals": {
+        "console": true,
+        "process.exitCode": "write",
+      },
+      "packages": {
+        "some-package>entropoetry>bn.js": true
+      }
+    },
+    "some-package>entropoetry>bn.js": {
+      "builtin": {
+        "buffer.Buffer": true
+      },
+      "globals": {
+        "Buffer": true
+      }
+    }
+  }
+}
+```
+
+## Fields
+### resources 
+
+All packages in your dependency graph accessible via `require()`.  
+Note that each resource is not represented just by the name of the package, but by a full dependency tree path to it. LavaMoat picks the shortest dependency path for each entry in resources to prevent a malicious package crafted to be represented with the same name but not fetched from npm to hijack permissions.
+
+
+### packages and builtin
+
+All packages accessible by the dependency. In this example, `some-package` has access to `entropoetry`.This means that `some-package` can `require('entropoetry')`.
+
+`entropoetry` can also `require('assert')` - a built-in module of Node.js.
+
+
+### globals
+ All platform APIs and global variables accessible by the dependency. In this example,`entropoetry` has access to `console` and it can write to process.exitCode. Global access is read-only, unless defined by the `write` value.
+
+Write access is represented by a string to make it stand out when inspecting the policy file.
+
+## Limiting access to fields of powerful objects
+
+You may have noticed the dot notation in keys within globals, packages and builtins. It can be used to limit the reference the package gets to just a subset of fields on the object.
+
+`"Buffer.from": true` means the `Buffer` global reference will exist in the package's scope but will only contain the `from` field and no other fields.  
+Same goes for `"buffer.Buffer": true` in builtins - when the package does `const buffer = require('buffer')` it will get an object with just one reference to `Buffer` constructor.
+
+LavaMoat can often generate precise lists of fields used by the package. You can narrow them down further if need be by creating a `policy-override.json` file next to the `policy.json`.
+
+# Using policy-override to expand or limit access
+
+## Adding access where LavaMoat's policy was too strict
+
+Policy files are generated by statically analyzing the source of packages and finding out which powerful references they seem to be using. It's impossible to detect it 100% correctly because there are JavaScript constructs that can't be detected without running the code.  
```

Comment:
What is a "powerful reference"?

What about something like this?

```
LavaMoat generates policy files by statically analyzing the source of a package and discovering references to variables and other packages (resources). However, LavaMoat is not perfect, and in certain cases, some references will not be visible until the code is executed, so they will not be added to the policy file automatically. This can lead to LavaMoat throwing an error at runtime as it will see an attempt to access an unpermitted resource. In this case, if you know that such a resource should be permitted, you can manually add it to the policy override file.
```

Reply:
powerful objects / powers are a key concept here. I might need to explain it more, but I'm not interested in removing.

Reply:
Cool, no problem!